holoviews/tests/ui/bokeh/test_datashader.py

```diff
@@ -0,0 +1,54 @@
+from functools import partial
+from unittest import SkipTest
+
+import pandas as pd
+import pytest
+
+import holoviews as hv
+from holoviews.operation import apply_when
+
+try:
+    from holoviews.operation.datashader import rasterize
+except ImportError:
+    raise SkipTest("Datashader not available")
+from ...plotting.utils import ParamLogStream
+from .. import expect, wait_until
+
+pytestmark = pytest.mark.ui
+
+
+@pytest.mark.usefixtures("bokeh_backend")
+def test_apply_when_zoom(serve_hv):
```

Comment:
I don't know why this would ever have to be a UI test.

Reply:
UI tests should only be implemented for functionality that actually happens on the frontend. In this case you can simulate all the conditions entirely in Python.